test/test_ffs.cpp

```diff
@@ -79,3 +79,26 @@ TORRENT_TEST(clz)
 		TEST_EQUAL(aux::clz({buf, 5}), t.second);
 	}
 }
+
+TORRENT_TEST(flz_u32)
+{
+	std::uint32_t v = 0;
+	TEST_EQUAL(aux::flz_sw(v), 31);
+	TEST_EQUAL(aux::flz_hw(v), 31);
+	TEST_EQUAL(aux::flz(v), 31);
+
+	v = 0xffffffff;
+	TEST_EQUAL(aux::flz_sw(v), -1);
+	TEST_EQUAL(aux::flz_hw(v), -1);
+	TEST_EQUAL(aux::flz(v), -1);
+
+	v = 0xff00ff00;
+	TEST_EQUAL(aux::flz_sw(v), 23);
```

Comment:
how come this doesn't return 31? if v is interpreted as big endian (assuming the test is run on a little-endian machine), the native value would be: 0x00ff00ff, and the highest index zero would be 31.

Speaking of which, these tests need to run htonl() on the values being assigned to v.